runtime/src/stakes.rs

```diff
@@ -116,16 +117,19 @@ impl Stakes {
         pubkey: &Pubkey,
         account: &Account,
         fix_stake_deactivate: bool,
+        check_vote_init: bool,
     ) -> Option<ArcVoteAccount> {
         if solana_vote_program::check_id(&account.owner) {
             // unconditionally remove existing at first; there is no dependent calculated state for
             // votes, not like stakes (stake codepath maintains calculated stake value grouped by
             // delegated vote pubkey)
             let old = self.vote_accounts.remove(pubkey);
-            // when account is removed (lamports == 0), don't readd so that given
-            // `pubkey` can be used for any owner in the future, while not
-            // affecting Stakes.
-            if account.lamports != 0 {
+            // when account is removed (lamports == 0 or data uninitialized), don't read so that
+            // given `pubkey` can be used for any owner in the future, while not affecting Stakes.
+            if account.lamports != 0
+                && !(check_vote_init && (account.data.len() != VoteState::size_of())
```

Comment:
Maybe move this size check into `is_unitialized_no_deser`?

Reply:
Good idea; makes the indexing more clear

Reply:
Adding a unit test for `is_unitialized_no_deser` also

Reply:
Updated